pkg/observable/channel/replay.go

```diff
@@ -111,7 +112,18 @@ func (ro *replayObservable[V]) Subscribe(ctx context.Context) observable.Observe
 	ro.replayBufferMu.RLock()
 	defer ro.replayBufferMu.RUnlock()
 
-	observer := NewObserver[V](ctx, ro.observerManager.remove)
+	if ctx == nil {
+		ctx = context.Background()
+	}
+
+	// caller can cancel context or close the publish channel to unsubscribe active observers
+	ctx, cancel := context.WithCancel(ctx)
+	removeAndCancel := func(toRemove observable.Observer[V]) {
+		ro.observerManager.remove(toRemove)
+		cancel()
```

Comment:
If the caller never calles `removeAndCancel`, does this never get cancelled?

Reply:
`removeAndCancel` is only ever called in the `Observer` implementation (see: `#unsubscribe()`, where it's referenced as `observer#onUnsubscribe()`) via [`channelObserverManager#goUnsubscribeOnDone()`](https://github.com/pokt-network/poktroll/blob/ddfdd3ca0331991e98590a1366bcdffc7deee109/pkg/observable/channel/observer_manager.go#L135), below.